File: windows/Services/RoutePersistenceService.cs
using Starsky.Desktop.Models;

namespace Starsky.Desktop.Services;

public class RoutePersistenceService(SettingsService settings)
{
	public List<SavedWindowState> GetRoutes() => settings.Current.Windows;

    public void SaveRoute(int index, string route, SavedWindowState? geometry = null)
    {
        var windows = settings.Current.Windows;
        while (windows.Count <= index)
        {
	        windows.Add(new SavedWindowState());
        }

        windows[index].Route = route;
        if (geometry != null)
        {
            windows[index].Left = geometry.Left;
            windows[index].Top = geometry.Top;
            windows[index].Width = geometry.Width;
            windows[index].Height = geometry.Height;
            windows[index].IsMaximized = geometry.IsMaximized;
        }

        settings.Save();
    }

    public void RemoveRoute(int index)
    {
        var windows = settings.Current.Windows;
        if (index >= 0 && index < windows.Count)
        {
            windows.RemoveAt(index);
            settings.Save();
        }
    }

    public void ClearAll()
    {
        settings.Current.Windows.Clear();
        settings.Save();
    }
}

File: mac/starsky/Services/RoutePersistenceService.swift
import Foundation

class RoutePersistenceService {
    private let settingsService: SettingsService

    init(settingsService: SettingsService) {
        self.settingsService = settingsService
    }

    func getRoutes() -> [SavedWindowState] {
        settingsService.current.windows
    }

    func saveRoute(index: Int, route: String, geometry: SavedWindowState? = nil) {
        var settings = settingsService.current
        while settings.windows.count <= index {
            settings.windows.append(SavedWindowState())
        }
        settings.windows[index].route = route
        if let g = geometry {
            settings.windows[index].x = g.x
            settings.windows[index].y = g.y
            settings.windows[index].width = g.width
            settings.windows[index].height = g.height
            settings.windows[index].isMaximized = g.isMaximized
        }
        settingsService.save(settings)
    }

    func removeRoute(index: Int) {
        var settings = settingsService.current
        guard index < settings.windows.count else { return }
        settings.windows.remove(at: index)
        settingsService.save(settings)
    }

    func clearAll() {
        var settings = settingsService.current
        settings.windows = []
        settingsService.save(settings)
    }
}
